src/Assets/Javascript/map.js

```diff
@@ -175,32 +209,171 @@ const initGoogleMaps = () => {
 
   const locations = window.locations
 
-  for (let i = 0, length = locations.length; i < length; i++) {
+  const postcodeRegex = /([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))\s?[0-9][A-Za-z]{2})/
+
+  const foundPostcodes = new Set()
+  const foundLocationsWithoutPostcode = new Set()
+
+  const extractPostcodeIfExists = addressToUse => {
+    const postcodeResults = postcodeRegex.exec(addressToUse)
+    const postcode = (postcodeResults || []).filter(Boolean).shift()
+    if (postcode) {
+      foundPostcodes.add(postcode)
+    } else {
+      foundLocationsWithoutPostcode.add(addressToUse)
+    }
+    return postcode ? postcode.replace(" ", "") : null
+  }
+
+  // pins = {
+  //   'lat,lng': {
+  //       distance: meters,
+  //       locations: []
+  //   }
+  // }
+  const pins = locations.reduce((pins, location) => {
```

Comment:
is this self invoke or something?

`const pins = locations.reduce((pins, location) => {`

I'm seeing pins as a variable and pins as a parameter

as well as 

`const pin = pins[key] || {}`

just sanity checking...